rskj-core/src/test/java/co/rsk/peg/RegisterBtcTransactionIT.java

```diff
@@ -0,0 +1,229 @@
+package co.rsk.peg;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+import co.rsk.bitcoinj.core.*;
+import co.rsk.bitcoinj.script.ScriptBuilder;
+import co.rsk.core.RskAddress;
+import co.rsk.bitcoinj.store.BlockStoreException;
+import co.rsk.net.utils.TransactionUtils;
+import co.rsk.peg.bitcoin.BitcoinTestUtils;
+import co.rsk.peg.btcLockSender.BtcLockSenderProvider;
+import co.rsk.peg.constants.BridgeConstants;
+import co.rsk.peg.constants.BridgeMainNetConstants;
+import co.rsk.peg.federation.*;
+import co.rsk.peg.federation.constants.FederationConstants;
+import co.rsk.peg.feeperkb.FeePerKbSupport;
+import co.rsk.peg.storage.BridgeStorageAccessorImpl;
+import co.rsk.peg.storage.StorageAccessor;
+import co.rsk.test.builders.BridgeSupportBuilder;
+import co.rsk.test.builders.FederationSupportBuilder;
+import org.bouncycastle.util.encoders.Hex;
+import org.ethereum.config.blockchain.upgrades.ActivationConfig;
+import org.ethereum.config.blockchain.upgrades.ActivationConfigsForTest;
+import org.ethereum.core.*;
+import org.ethereum.vm.PrecompiledContracts;
+import org.junit.jupiter.api.Test;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.util.*;
+
+
+public class RegisterBtcTransactionIT {
+
+    private final BridgeConstants bridgeConstants = BridgeMainNetConstants.getInstance();
+    private final NetworkParameters btcParams = bridgeConstants.getBtcParams();
+    private final BridgeSupportBuilder bridgeSupportBuilder = BridgeSupportBuilder.builder();
+
+    @Test
+    void whenRegisterALegacyBtcTransaction_shouldRegisterTheNewUtxoAndTransferTheRbtcBalance() {
+        // Arrange
+        ActivationConfig.ForBlock activationConfig = ActivationConfigsForTest.all().forBlock(0);
+        Repository repository = BridgeSupportTestUtil.createRepository();
+        Repository track = repository.startTracking();
+        Block rskExecutionBlock = getRskExecutionBlock();
+
+        BtcLockSenderProvider btcLockSenderProvider = new BtcLockSenderProvider();
+        FeePerKbSupport feePerKbSupport = getFeePerKbSupport();
+
+        Federation federation = P2shErpFederationBuilder.builder().build();
+        FederationStorageProvider federationStorageProvider = getFederationStorageProvider(track, federation);
+        FederationSupport federationSupport = getFederationSupport(federationStorageProvider, activationConfig, bridgeConstants.getFederationConstants());
+
+        BtcECKey btcPublicKey = new BtcECKey();
+        Coin btcTransferred = Coin.COIN;
```

Comment:
Let's use a more common/realistic value, like `minimumPegin`, wdyt?

Reply:
I thought COIN was normally used as regular value 
https://github.com/rsksmart/rskj/blob/4fbdf6978d7aeb12137b6baed903e96cfe6eab2b/rskj-core/src/test/java/co/rsk/peg/BridgeSupportAddSignatureTest.java#L544

Reply:
Well, on Mainnet, I don't think many users will send 1 BTC. The most common value I think, will be close to the minimum pegin value.

Reply:
perfecto, changed already!